### PR TITLE
Using Result instead of tuple where applicable

### PR DESCRIPTION
**Pull request by acn-sbuad**
## Description

Removed use of tuples to return success or error response. 

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [x] All tests run green

### CODE REVIEW

File: src/Altinn.Notifications.Core/Services/OrderRequestService.cs [hunk 1]
```diff
@@ -33,7 +33,7 @@ public OrderRequestService(IOrderRepository repository, IGuidService guid, IDate
     }
 
     /// <inheritdoc/>
-    public async Task<(NotificationOrder? Order, ServiceError? Error)> RegisterNotificationOrder(NotificationOrderRequest orderRequest)
+    public async Task<Result<NotificationOrder, ServiceError>> RegisterNotificationOrder(NotificationOrderRequest orderRequest)
```
Comment (tba76) [inline src/Altinn.Notifications.Core/Services/OrderRequestService.cs:36]: Do we need ServiceError here, it will never be returned?

Comment (acn-sbuad) [reply src/Altinn.Notifications.Core/Services/OrderRequestService.cs:36]: Saving order to db could fail, but we don't catch the exception for now. So either we catch exceptions from the repository layer and return service error or drop the service error.

Comment (tba76) [reply src/Altinn.Notifications.Core/Services/OrderRequestService.cs]: Drop is good :)

**Review by tba76 — APPROVED** [review]
🍺

**Pull request merged**